datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -1114,16 +1119,24 @@ private void handleLeaderDoAssignment(boolean cleanUpOrphanNodes) {
       // it failed to create or delete zNodes), we will do our best to continue the current process
       // and schedule a retry. The retry should be able to diff the remaining ZooKeeper work
       _adapter.updateAllAssignments(newAssignmentsByInstance);
+
+      stoppingDatastreamGroups.forEach(datastreamGroup -> datastreamGroup.getDatastreams().forEach(datastream -> {
```

Comment:
For readability purpose, it is a good practice to have 1 for loop per line. I understand that `forEach` has its benefits, but in this case, a regular for loop will avoid the use of `AtomicInteger` which is an over-kill.